src/mnemonic-key.ts

```diff
@@ -16,6 +17,28 @@ export class MnemonicKey extends Key {
       mnemonic = Wallet.generateMnemonic()
     }
 
-    this.addByMnemonic(mnemonic, options.index || 0)
+    this._addByMnemonic(mnemonic, options.index || 0, options.defaultPath, options.addresses)
+  }
+
+  private _addByMnemonic(phrase: string, index = 0, defaultPath = '1023', addresses = 1) {
+    if (!HDNode.isValidMnemonic(phrase)) {
+      throw new Error(`Invalid mnemonic phrase: ${phrase}`)
+    }
+    const privateKeys: string[] = []
+
+    const seed = bip39.mnemonicToSeed(phrase)
+    const hdKey = hdkey.fromMasterSeed(seed)
+    const path = this.messenger.chainType === ChainType.Harmony ? '1023' : defaultPath
+
+    for (let i = 0; i < addresses; i++) {
+      const childKey =
+        path !== '1023' || index !== 0
+          ? hdKey.derive(`m/44'/${path}'/0'/0/${index + i}`)
+          : hdKey.derive(`${HDPath}${index}`)
+      const privateKey = `0x${childKey.privateKey.toString('hex')}`
+      privateKeys.push(privateKey)
```

Comment:
Not need for this, you need to add all the keys with `this.addByPrivateKey(privateKey)`, because just the first one will be use as signer.

Reply:
Okay 🤔, I'll change it right now...